Blogpost about the State of React Native Community in 2018 (#719)
facebook/react-native-website: website/blog/2018-12-21-state-of-react-native-community.md

---
title: The State the React Native Community in 2018
author: Lorenzo Sciandra
authorTitle: Core Maintainer & React Native Developer
authorURL: https://github.com/kelset
authorImageURL: https://avatars2.githubusercontent.com/u/16104054?s=460&v=4
authorTwitter: kelset
category: announcements
---

> Two years from now, we’ll probably look back at 2018 as a turning point in the React Native development history.

Reviewer: This sounds a bit off to me. What do you think about "We made a large number of changes about how we develop and communicate about React Native. We hope that in two years from now we can look back and see that 2018 was a turning point for React Native."?

Author: I’m not sure I understand what is off there, but ok 👍

> This year we made a large number of changes about how we develop and communicate about React Native; we hope that in two years from now we can look back and see that this shift was a turning point for React Native.

Reviewer 2: ```suggestion
In 2018 the React Native Community made a number of changes about how we develop and communicate about React Native. We believe that a few years from now we will look back and see that this shift was a turning point for React Native.
```

- Specify the year
- Remove "large" (how do you quantify this)
- Remove semicolon
- Change hope to belief
- Change two years to a few years

> A lot of eyes are currently pointed at the incoming rewrite of the architecture, widely known as [Fabric](https://github.com/react-native-community/discussions-and-proposals/issues/4), which will enable our mobile apps to reach new peaks of mobile performances (alongside JSI & TurboModules, which are equally as important to reach said goal).

Reviewer: "reach new peaks of mobile performances" sounds a bit wrong, because you want low performance. Fabric is also not only there to fix performance problems, but other fundamental flaws in RN's architecture. I would probably recommend rewording this paragraph like this:

> A lot of people are excited about the rewrite of React Native's architecture, widely known as [Fabric](https://github.com/react-native-community/discussions-and-proposals/issues/4). Among other things, this will fix fundamental flaws in React Native's architecture and will set up React Native for success in the future together with JSI and TurboModules (+ add links pls).

Author: I’m not sure I understand here - low performance is the goal? Isn’t ‘high’ the right adjective? I’ll ask around.

Reviewer 3: The suggested wording is 👌, but “low performance” means “suboptimal”

Reviewer 4: haha probably just a mistakes XD
i'd never want low performance

Reviewer: Oops, yeah you are right. Sorry I had a baby crawling around me while I wrote this! :D

I meant you want lower wait times but obviously higher performance. Peak performance works but I think “new peaks of performance” sounds off.

> But I will argue that the biggest shift initiated during 2018 is not code related;  this title should go to the empowerment that the React Native Community gained.

Reviewer: You don't have to argue, just say it as it is: "The biggest shift initiated in 2018 was to empower the React Native Community: …"

Author: 👌

> A lot of people are excited about the rewrite of React Native's architecture, widely known as [Fabric](https://github.com/react-native-community/discussions-and-proposals/issues/4). Among other things, this will fix fundamental flaws in React Native's architecture and will set up React Native for success in the future together with [JSI and TurboModules](https://github.com/react-native-community/discussions-and-proposals/issues/40).

Reviewer 2: ```suggestion
A lot of people are excited about the rewrite of React Native's architecture, widely known as [Fabric](https://github.com/react-native-community/discussions-and-proposals/issues/4). Among other things, this will fix fundamental limitations in React Native's architecture and will set up React Native for success in the future together with [JSI and TurboModules](https://github.com/react-native-community/discussions-and-proposals/issues/40).
```

- Switch "flaws" to "limitations"

> Since the very beginning, Facebook allowed developers from all around the world to participate in this Open Source project; investing a few super active members of the community of responsibilities like owning the release cycle.

Reviewer: "From the beginning Facebook encouraged developers from all around the world to participate in React Native's open source project. Since then, a number of core contributors emerged who handle, among other things, the release process" ?

Author: 👌

> The biggest shift initiated in 2018 was to empower the React Native Community: from the beginning Facebook encouraged developers from all around the world to participate in React Native's open source project. Since then, a number of core contributors emerged who handle, among other things, the release process.

Reviewer 2: ```suggestion
The biggest shift in 2018 was to empower the React Native Community. From the beginning, Facebook encouraged developers from all around the world to participate in React Native's open source project. Since then, a number of core contributors emerged who handle, among other things, the release process.
```

- Initiated is redundant
- Remove colon 
- Add comma for introductory sentence

> And, this year, the members of the Core Contributors group took a few substantial steps towards making the whole community more able to shape the future of this awesome project: two repositories have been created throughout the year precisely with this goal in mind.

Reviewer: "This year […]", drop the "and".

"towards allowing the community to shape the future of the project" – drop the awesome, we don't need to celebrate ourselves too much here ;)

Reviewer: End this paragraph with a colon (":") because you are enumerating the two changes.

Author: 👌

> This year, these members took a few substantial steps towards making the whole community more able to shape the future of this project; two repositories have been created throughout the year precisely with this goal in mind:

Reviewer 2: ```suggestion
These members took a few substantial steps towards making the whole community more empowered to shape the future of this project. Two repositories have been created throughout the year precisely with this goal in mind:
```

- "This year" is the context of the entire post, no need to restate again
- Change "more able" to "more empowered"
- Remote semi-colon

## [`react-native-releases`](https://github.com/react-native-community/react-native-releases) 📬

> This repository, started around Feb, served the dual purpose of allowing everyone to keep up the new releases in a more collaborative manner and opened the conversation of what would be part of a certain release to whoever wanted to suggest a cherry-pick (like for [0.57.8](https://github.com/react-native-community/react-native-releases/issues/71) and all its previous versions).

Reviewer: "serves the dual purpose"

Author: 👌

> This repository, started around Feb, serves the dual purpose of allowing everyone to keep up the new releases in a more collaborative manner and opened the conversation of what would be part of a certain release to whoever wanted to suggest a cherry-pick (like for [0.57.8](https://github.com/react-native-community/react-native-releases/issues/71) and all its previous versions).

Reviewer 2: ```suggestion
This repository, created in January, serves the dual purpose of allowing everyone to keep up the new releases in a more collaborative manner and opened the conversation of what would be part of a certain release to whomever wanted to suggest a cherry-pick (like for [0.57.8](https://github.com/react-native-community/react-native-releases/issues/71) and all its previous versions).
```

- Change from "started" to "created" to mirror language for the other repository 
- Change to January (which is when the initial commit was pushed)
- I believe whomever is the correct word here

This has been the driving force behind moving away from a monthly release cycle, and the "long term support" approach currently used for version 0.57.x.

Half of the merit in reaching these decisions goes to the other repository created this year:

## [`discussions-and-proposals`](https://github.com/react-native-community/discussions-and-proposals) 🗣

> This other ‘meta’ repository, created a few months after its sister (in July), expanded on the idea of a more open environment for conversations on React Native; while, partially, this need was covered by a few issues labelled [`For Discussion`](https://github.com/facebook/react-native/labels/For%20Discussion) in the main repository, we wanted to give our own spin to the RFC approach that other libraries (like React) have.

Reviewer: I would drop the "sister" thing. Just say "This repository was created in July"

Author: 👌

> This other ‘meta’ repository, created in July, expanded on the idea of a more open environment for conversations on React Native; while, partially, this need was covered by a few issues labelled [`For Discussion`](https://github.com/facebook/react-native/labels/For%20Discussion) in the main repository, we wanted to give our own spin to the RFC approach that other libraries (like React) have.

Reviewer 2: ```suggestion
This repository, created in July, expanded on the idea of a more open environment for conversations on React Native. Previously, this need was handled by issues labelled [`For Discussion`](https://github.com/facebook/react-native/labels/For%20Discussion) in the main repository, but we wanted to expand this strategy to an RFC approach that other libraries have (e.g. React).
```

- Cleaned up to make it less conversational

> This experiment found immediately its role in the React Native lifecycle: even the FB team is now relying heavily on it to discuss what could [be improved in React Native](https://github.com/react-native-community/discussions-and-proposals/issues/64), and coordinate the efforts around the [Lean Core project](https://github.com/react-native-community/discussions-and-proposals/issues/6) - among other interesting conversations.

Reviewer 2: ```suggestion
This experiment immediately found its role in the React Native lifecycle. The Facebook team is now relying heavily on the RFC process to discuss what could [be improved in React Native](https://github.com/react-native-community/discussions-and-proposals/issues/64), and coordinate the efforts around the [Lean Core project](https://github.com/react-native-community/discussions-and-proposals/issues/6) - among other interesting discussions.
```

- Put the adverb first
- Remove colon
- Remove "even" from "even FB"
- Spell Facebook
- Specify the pronoun because context isn't clear
- Use "discussions" for consistent language

> But we are far from done.

Reviewer 2: ```suggestion
```

- This is implied and not necessary IMO

### What awaits ahead 🎓

> As explained above, over the past 6-8 months, the [React Native Community organisation](https://github.com/react-native-community) has been used by core contributors to take more ownership over the development of React Native, and enhance the collaboration with Facebook. But this always lacked the formal structure that similar projects may have in place. 

Reviewer 2: ```suggestion
Over the past 6-8 months, the [React Native Community GitHub organization](https://github.com/react-native-community) was created and used by core contributors to take more ownership over the development of React Native, and enhance the collaboration with Facebook. But this always lacked the formal structure that similar projects may have in place. 
```

- No need to say you explained it above because you explained it above
- Specify you're talking about a GitHub org
- Say that it was created

> As explained above, over the past 6-8 months, the [React Native Community organisation](https://github.com/react-native-community) has become de-facto "officially endorsed by FB" - but it always lacked the formal structure that similar projects (like Flutter or Ionic) have for their plugin libraries. By enforcing a set of standards for all the packages/repos hosted in it, providing a common place for all the maintainers to help each other and providing quality code, this organisation can set the example for everyone in the larger developer community.

Reviewer: I'm not really sure it makes sense here to specifically call out alternative frameworks here? Can we drop them?

I'm also not sure about "official endorsed by FB" in quotes. It sounds a bit odd. Could we just say "The React Native core contributors are taking more ownership over the development of React Native, and are collaborating with Facebook more than ever before"?

Author: I’ve changed it to be closer to your version, but I still think that it’s important to communicate that we are not ‘blind’ (as community around RN) to the other projects and that we have the sensibility to tell how and what is good, and use that to inspire positive change in our ‘world’.

Reviewer: I understand what you are saying. I’m just pointing out that those projects aren’t the motivation for us to improve React Native, it’s purely coincidental that they have been doing similar things.

Reviewer 2: I don't think this sentence really adds anything, what's the messaging you're after?

> By enforcing a set of standards for all the packages/repos hosted in it, providing a common place for all the maintainers to help each other and providing quality code, this organisation can set the example for everyone in the larger developer community.

Reviewer 2: ```suggestion
This organization can set the example for everyone in the larger developer community by enforcing a set of standards for all the packages/repos hosted in it, providing a common place for all the maintainers to help each other and providing quality code.
```

- Use active voice

> We are planning, for the early 2019, to have this new set of guidelines in place - you can let us know what you think in the [dedicated discussion](https://github.com/react-native-community/discussions-and-proposals/issues/63), and in the manifesto proposal that will be written soon.

Reviewer 2: ```suggestion
In early 2019, we will have this new set of guidelines in place. Let us know what you think in the [dedicated discussion](https://github.com/react-native-community/discussions-and-proposals/issues/63).
```

- Planning is implied, it's in the future
- Remove em dash
- Remove mention of something doesn't have any mention/context elsewhere in the article and doesn't exist yet

> With this last founding stone, we are confident that the community will become more inclusive and when 1.0 will be reached, we will all be able to write (even more) awesome apps, capable of leveraging this joined effort 🤗

Reviewer 2: ```suggestion
We are confident that with these changes, the community will become more collaborative so that when we reach 1.0, we will all continue to write (even more) awesome apps by leveraging this joined effort 🤗
```

I would probably remove this bit altogether

----

> I hope you are as excited as I am in looking at the future of this community, and that I will see all of you involved either in the conversations happening in the repositories listed above - or via the awesome code you’ll produce.

Reviewer 2: ```suggestion
I hope you are as excited as we are about the future of this community. We're excited to see all of you involved either in the conversations happening in the repositories listed above or via the awesome code you’ll produce.
```

- Switch "I" to "we" since the whole article is written in the "we" voice so far
- Add sentence structure

> We are aware that our approach to communicate these efforts has not been as effective as we liked, and in an attempt to give you all an easier time keeping up with everything going on in the React Native Community (from releases to active discussions) we created a new twitter account that you can rely on [@ReactNativeComm](https://twitter.com/ReactNativeComm) (if you are not on it remember that you can always watch repositories via GitHub). It will become more actively starting 2019, but feel free to start following it right away 🐣

Reviewer 2: This seems added in as an afterthought, the previous paragraph seems to be the closing paragraph. May consider a better place to add this in

Author: I see what you are saying, but tbh I think it's ok to put it at the end because it's basically "Off Topic" and here it's the 'least bad' spot to have this communicated 😅

Reviewer 2: What if it was one of the three header points, added to line 32:

```

## [@ReactNativeComm](https://twitter.com/ReactNativeComm)`

<move this paragraph here>
```

That seems inline with the lead-in sentence for these points: 

> These members took a few substantial steps towards making the whole community more empowered to shape the future of this project. _They taken these steps with this goal in mind_[edited]:

And this would promote it to "on topic" :D

Author: that's a good idea! I'll edit

> Happy holidays & new year to you all 🎅

Reviewer 2: ```suggestion
```

- Will probably be outdated by the time this is posted
